### ISSUE TITLE
Nullable kotlin records break columns with default values

### ISSUE THREAD

**Issue by serega6531** [jOOQ/jOOQ#14817]
### Expected behavior

The query executes without errors (actual names hidden for privacy)
```sql
insert into "public"."censored_table" ("id", "column2")
values (1, 2);
```

### Actual behavior

`ERROR: null value in column "censored_column" of relation "censored_table" violates not-null constraint`
```sql
insert into "public"."censored_table" ("id", "column2", "censored_column")
values (1, 2, NULL);
```

### Steps to reproduce the problem

1. `kotlinNotNullRecordAttributes = true`
2. Have a table with a column that has a default value.
```sql
create table censored_table
(
    id numeric not null primary key,
    column2 numeric not null,
    censored_column boolean default true not null
);
```
Field from the table:
```kotlin
    val CENSORED_COLUMN: TableField<..., Boolean?> = createField(DSL.name("censored_column"), SQLDataType.BOOLEAN.nullable(false).defaultValue(DSL.field(DSL.raw("true"), SQLDataType.BOOLEAN)), this, "")

```
Record for that table:
```kotlin
    open var id: Int
        set(value): Unit = set(0, value)
        get(): Int = get(0) as Int

    open var column2: Int
        set(value): Unit = set(1, value)
        get(): Int = get(1) as Int

    open var censoredColumn: Boolean?
        set(value): Unit = set(2, value)
        get(): Boolean? = get(2) as Boolean?
```
Note how the type is `Boolean?` even though the column is not-null (could also be a bug).
3. Create and insert a record (don't fill the third column)
```kotlin
val test = CensoredTableRecord(
    id = 1,
    column2 = 2
)

dsl.batchInsert(test);
```

### jOOQ Version

OOS 3.18.0

### Database product and version

PostgreSQL 13.3 on aarch64-unknown-linux-musl, compiled by gcc (Alpine 10.3.1_git20210424) 10.3.1 20210424, 64-bit

### Java Version

openjdk version "18.0.2.1" 2022-08-18

**Comment by lukaseder** [comment]
Thanks for your report. I can confirm the problem. Will investigate right away.

**Comment by lukaseder** [comment]
Ah yes, it's a problem with the generated constructor, where omitting the argument leads to a `NULL` value being set (the constructor default).

It's probably the same issue as these ones: 

- https://github.com/jOOQ/jOOQ/issues/13821
- https://github.com/jOOQ/jOOQ/issues/14727

The latter was fixed in jOOQ 3.18.0. I'd argue that this issue here should be fixed for the same reasons as the latter. Throughout jOOQ, it can be expected that setting a `NULL` value on a record attribute that is *both* of these, should be ignored:

- `NOT NULL`
- `DEFAULT`-ed

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.19.0 and 3.18.1 (https://github.com/jOOQ/jOOQ/issues/14822).

Further backports aren't planned because of the subtle change of behaviour. It is backported to 3.18.1 for consistency reasons with #14727

**Comment by serega6531** [comment]
Thank you! Do you know if the type being nullable is expected behaviour in this case?

**Comment by lukaseder** [comment]
> Further backports aren't planned because of the subtle change of behaviour. It is backported to 3.18.1 for consistency reasons with #14727

Good thing I didn't backport this any further. Here's a subtle change of behaviour that produced a regression when upgrading beyond 3.18.1:

- https://github.com/jOOQ/jOOQ/issues/17224

It will be fixed in 3.20.0, 3.19.12, 3.18.19
